core/src/main/java/com/netflix/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -109,32 +104,30 @@ public void commit(TableMetadata base, TableMetadata metadata) {
       return;
     }
 
+    FileIO newMetadataIO = io(metadata);
```

Comment:
Why move this before the validation checks? If those fail, this file would need to be cleaned up.